distribution/ddf-common/src/main/resources/bin/setenv

```diff
@@ -59,9 +59,10 @@ export EXTRA_JAVA_OPTS
 #
 
 # export JAVA_MIN_MEM # Minimum memory for the JVM
-export JAVA_MAX_MEM=4096M
+# export JAVA_MAX_MEM=4096M
 # export KARAF_HOME # Karaf home folder
 # export KARAF_DATA # Karaf data folder
 # export KARAF_BASE # Karaf base folder
 
 export KARAF_OPTS="-Dfile.encoding=UTF8 -Dddf.home=$DDF_HOME"
+export JAVA_OPTS=-"server -Xms128M -Xmx4096M -XX:+UnlockDiagnosticVMOptions -XX:+UnsyncloadClass -XX:+DisableAttachMechanism"
```

Comment:
We could start with 2G, but as soon as heap requirements force it to grow, the JVM will shift things about to grow the heap. It's a long-standing server optimization to set those two values to the same size so you pre-allocate all you need and never need to reorg the heap. It's not a requirement certainly, and I could see an argument for a 2G min, but 128M (or the tiny defaults when not specified) would be _bad_.